test/src/test/java/org/corfudb/integration/StreamingIT.java

```diff
@@ -812,4 +854,166 @@ public void testStreamingMultiTableSingleListener() throws Exception {
 
         assertThat(shutdownCorfuServer(corfuServer)).isTrue();
     }
+
+    /**
+     * Test the case where streaming is interrupted due to an unrecoverable
+     * TrimmedException.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testStreamingWithTrimmedException() throws Exception {
+        // Run a corfu server & initialize CorfuStore
+        initializeCorfu();
+
+        // Record the initial timestamp.
+        Timestamp ts1 = store.getTimestamp();
+
+        final int totalUpdates = 100;
+
+        writeUpdatesToDefaultTable(totalUpdates);
+
+        // Checkpoint and Trim
+        Token trimPoint = checkpointAndTrim(namespace, Arrays.asList(defaultTableName));
+
+        // Subscribe listener to stream tag 'sample_streamer_1'
+        final CountDownLatch errorListener = new CountDownLatch(1);
+
+        // Start Listener on Trimmed Space
+        TrimExceptionListener listener = new TrimExceptionListener("stream_listener", errorListener);
+        store.subscribeListener(listener, namespace, defaultTag,
+                Collections.singletonList(defaultTableName), ts1);
+
+        // Wait until TrimmedException is received by listener
+        errorListener.await();
+
+        // Confirm we are unsubscribed (subscribing again does not throw an error) -> subscribing beyond trim point
+        Timestamp trimTimestamp = Timestamp.newBuilder()
+                .setEpoch(trimPoint.getEpoch())
+                .setSequence(trimPoint.getSequence())
+                .build();
+        store.subscribeListener(listener, namespace, "sample_streamer_1",
+                Collections.singletonList(defaultTableName), trimTimestamp);
+
+        // Verify number of updates received by listener is zero (empty) as the space was trimmed.
+        assertThat(listener.getUpdates()).isEmpty();
```

Comment:
Also verify that CountDownLatch `errorListener` is decremented to 0?

Reply:
Not really needed right, as in L. 888 we blocked until the latch count down reached zero.